x-pack/plugins/security_solution/public/management/pages/policy/view/host_isolation_exceptions/components/delete_modal.tsx

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiCallOut, EuiConfirmModal, EuiSpacer, EuiText } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { ExceptionListItemSchema } from '@kbn/securitysolution-io-ts-list-types';
+import React from 'react';
+import { useMutation, useQueryClient } from 'react-query';
+import { useHttp, useToasts } from '../../../../../../common/lib/kibana';
+import { ServerApiError } from '../../../../../../common/types';
+import { updateOneHostIsolationExceptionItem } from '../../../../host_isolation_exceptions/service';
+
+export const PolicyHostIsolationExceptionsDeleteModal = ({
+  policyId,
+  exception,
+  onCancel,
+}: {
+  policyId: string;
+  exception: ExceptionListItemSchema;
+  onCancel: () => void;
+}) => {
+  const toasts = useToasts();
+  const http = useHttp();
+  const queryClient = useQueryClient();
+
+  const onDeleteError = (error: ServerApiError) => {
+    toasts.addError(error as unknown as Error, {
+      title: i18n.translate(
+        'xpack.securitySolution.endpoint.policy.hostIsolationExceptions.list.removeDialog.errorToastTitle',
+        {
+          defaultMessage: 'Error while attempt to remove host isolation exception',
+        }
+      ),
+    });
+    onCancel();
+  };
+
+  const onDeleteSuccess = () => {
+    queryClient.invalidateQueries(['endpointSpecificPolicies']);
+    queryClient.invalidateQueries(['hostIsolationExceptions']);
+    toasts.addSuccess({
+      title: i18n.translate(
+        'xpack.securitySolution.endpoint.policy.hostIsolationExceptions.list.removeDialog.successToastTitle',
+        { defaultMessage: 'Successfully removed' }
+      ),
+      text: i18n.translate(
+        'xpack.securitySolution.endpoint.policy.hostIsolationExceptions.list.removeDialog.successToastText',
+        {
+          defaultMessage: '"{exception}" has been removed from policy',
+          values: { exception: exception.name },
+        }
+      ),
+    });
+    onCancel();
+  };
+
+  const mutation = useMutation(
+    async () => {
+      const modifiedException = {
+        ...exception,
+        tags: exception.tags.filter((tag) => tag !== `policy:${policyId}`),
+      };
+      return updateOneHostIsolationExceptionItem(http, modifiedException);
+    },
+    {
+      onSuccess: onDeleteSuccess,
+      onError: onDeleteError,
+    }
+  );
+
+  const handleModalConfirm = () => {
+    mutation.mutate();
```

Comment:
nit: do this only if is not already done (!mutation.isLoading)

Reply:
This is handled already by the `isLoading` logic in the confirm EUI Component. It would add extra protection but it is not really necessary.